assets/js/editor-components/product-tag-control/index.tsx

```diff
@@ -0,0 +1,142 @@
+/**
+ * External dependencies
+ */
+import { __, _n, sprintf } from '@wordpress/i18n';
+import { useState, useEffect, useCallback, useMemo } from '@wordpress/element';
+import { SearchListControl } from '@woocommerce/editor-components/search-list-control';
+import { SelectControl } from '@wordpress/components';
+import { getSetting } from '@woocommerce/settings';
+import { useDebouncedCallback } from 'use-debounce';
+
+/**
+ * Internal dependencies
+ */
+import type { SearchListItem as SearchListItemProps } from '../search-list-control/types';
+import ProductTagItem from './product-tag-item';
+import type { ProductTagControlProps } from './types';
+import { getProductTags } from '../utils';
+import './style.scss';
+
+/**
+ * Component to handle searching and selecting product tags.
+ */
+const ProductTagControl = ( {
+	isCompact = false,
+	onChange,
+	onOperatorChange,
+	operator = 'any',
+	selected,
+}: ProductTagControlProps ): JSX.Element => {
+	const [ list, setList ] = useState< SearchListItemProps[] >( [] );
+	const [ loading, setLoading ] = useState( true );
+	const [ isMounted, setIsMounted ] = useState( false );
+	const limitTags = getSetting( 'limitTags', false );
+
+	const selectedTags = useMemo< SearchListItemProps[] >( () => {
+		return list.filter( ( item ) => selected.includes( item.id ) );
+	}, [ list, selected ] );
+
+	const onSearch = useCallback(
+		( search: string ) => {
+			setLoading( true );
+			getProductTags( { selected, search } )
+				.then( ( newList ) => {
+					setList( newList );
+					setLoading( false );
+				} )
+				.catch( () => {
+					setLoading( false );
+				} );
+		},
+		[ selected ]
+	);
+
+	// Load on mount.
+	useEffect( () => {
+		if ( isMounted ) {
+			return;
+		}
+		onSearch( '' );
+		setIsMounted( true );
+	}, [ onSearch, isMounted ] );
```

Comment:
This is a unique approach to executing `useEffect` just once. Wouldn't providing an empty array achieve the same outcome? This way, we could eliminate the need for the `isMounted` state.

Reply:
That won't work here because of the dependencies of the hook. Without the mounted check it would run on other renders if `onSearch` changed (it did in testing). We've used this pattern before.